### PR TITLE
feat(all): update packages, tests, links

### PR DESCRIPTION
**Pull request by m1ga**
* clean analytics file (still kept it since I'm not sure if other files use it)
* clean auth file (still kept it since I'm not sure if other files use it)
* fixed link to Oracle download and added OpenJDK link
* updated node packages
* removed auth/analytics test
* fixed some of the warnings in the tests
* added `allowEmptyCatch` rule to ignore empty blocks

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,28 +15,28 @@
...
+    "async": "^3.2.4",
     "colors": "1.4.0",
     "fs-extra": "~9.1.0",
     "request": "~2.88.0",
-    "semver": "~7.3.5",
+    "semver": "~7.3.8",
     "sprintf": "^0.1.5",
     "temp": "~0.9.4",
-    "uuid": "~8.3.2",
+    "uuid": "~9.0.0",
```
Comment (cb1kenobi) [inline package.json:26]: I'd have to test this, but I'm hesitant to blind merge a major bump.

File: lib/analytics.js [hunk 2]
```diff
@@ -11,340 +11,14 @@
...
-exports.addEvent = function addEvent(name, data, type) {
-	analyticsEvents.push({
-		id: uuid.v4(),
-		type: type || 'app.feature',
-		name: name,
-		ts: timestamp(),
-		data: data
-	});
+exports.addEvent = function addEvent(_name, _data, _type) {
+	analyticsEvents.push({});
```
Comment (cb1kenobi) [inline lib/analytics.js:19]: This function could be a noop. I don't think we need to populate the events array. With that said, it's harmless.

File: lib/auth.js [hunk 3]
```diff
@@ -13,306 +13,18 @@
...
-			};
-		} catch (e) { // Invalid session file. This should never happen
-			result = createLoggedOutSessionFile(sessionFile);
-		}
-	} else {
-		result = createLoggedOutSessionFile(sessionFile); // No prior history, create a new logged out file
-	}
-
-	return cachedStatus = result;
+exports.login = function login(_args) {};
```
Comment (cb1kenobi) [inline lib/auth.js:16]: Elsewhere you had to put a comment in the body of a scope to I believe make lint happy. I'm curious why linting complained elsewhere, but not here? FWIW, I do prefer the empty body as you have it.

**Review by cb1kenobi — CHANGES_REQUESTED** [review]
This PR is a great first step! Thanks for working on this.

File: lib/analytics.js (hunk 2 above)
Comment (m1ga) [reply lib/analytics.js]: Ah right. I had to use it for the test file, but since that is gone I can remove it :-)

File: lib/auth.js (hunk 3 above)
Comment (m1ga) [reply lib/auth.js:16]: I prefer this too, even for the code blocks in an if or catch. We could add `allowEmptyCatch` to ignore this warning

Comment (cb1kenobi) [reply lib/auth.js:16]: Yeah, I say we add `allowEmptyCatch` (which I didn't know existed!)

File: package.json (hunk 1 above)
Comment (m1ga) [reply package.json:26]: only tested it on Linux. btw: I don't see where it is used

File: lib/auth.js (hunk 3 above)
Comment (m1ga) [reply lib/auth.js:16]: done :+1:

**Review by cb1kenobi — APPROVED** [review]
LGTM

**Pull request merged**